app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
+      persistent: true,
+      connect_timeout: TIMEOUT)
+  end
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    route = 'spamreports.get.json'
+    query = { email: email }
+
+    response = get_request(route, query)
+
+    response.any?
+  rescue => e
```

Comment:
I agree with doing that generally, although I can see why the old code did it which I ported over (there are specs for catching `StandardError`).

By catching all it makes the client less likely to raise an error, and given we use it in the request cycle it seems to be a good idea in this case. Downside that there might be some errors caused by the app to be swallowed and go unnoticed, to mitigate this I could report any unexpected error to sentry? An unexpected error would be anything that is not an Excon or SendgridApi::Error.